src/task.rs

```diff
@@ -395,6 +395,23 @@ impl<T> Task<T> {
         let header = ptr as *const Header;
         unsafe { &*header }
     }
+
+    /// Get the current state of the task.
+    ///
+    /// Note that in a multithreaded environment, this state can change immediately after calling this function.
+    pub fn state(&self) -> TaskState {
+        let ptr = self.ptr.as_ptr();
+        let header = ptr as *const Header;
+
+        unsafe {
+            let state = (*header).state.load(Ordering::Acquire);
+            if state & (CLOSED | COMPLETED) != 0 {
+                TaskState::Completed
+            } else {
+                TaskState::Running
+            }
```

Comment:
Considering that `TaskState` is `non_exhaustive`, the user can really only rely on `state==Completed` (since `state==Running` may change its meaning in the future).

So I am starting to think that it might actually be preferable to have a method that returns a bool like `is_completed`.

Reply:
std JoinHandle has a similar unstable feature, which appears to be planned to be renamed is_finished.
https://github.com/rust-lang/rust/pull/94549